pymatbridge/tests/test_magic.py

```diff
@@ -82,7 +82,7 @@ def test_matrix(self):
     # Matlab struct type should be converted to a Python dict
     def test_struct(self):
         self.ip.run_cell('num = 2.567')
-        self.ip.run_cell('num_array = np.array([1.2,3.4,5.6])')
+        self.ip.run_cell('num_array = np.array([1.2,3.4,5.6]).reshape(3,1)')
```

Comment:
I think that round-tripping is important. If you send in an object of some
sort, and pull it back out, you don't expect it to change. I am not sure
how to apply this generally, though.

On Mon, Jan 19, 2015 at 5:18 PM, Steven Silvester <notifications@github.com>
wrote:

> In pymatbridge/tests/test_magic.py
> <https://github.com/arokem/python-matlab-bridge/pull/103#discussion-diff-23197883>
> :
>
> > @@ -82,7 +82,7 @@ def test_matrix(self):
> >      # Matlab struct type should be converted to a Python dict
> >      def test_struct(self):
> >          self.ip.run_cell('num = 2.567')
> > -        self.ip.run_cell('num_array = np.array([1.2,3.4,5.6])')
> > +        self.ip.run_cell('num_array = np.array([1.2,3.4,5.6]).reshape(3,1)')
>
> The original array is (3,) and it is sent and comes back as (3,1). The
> options are: call squeeze in decode_pymat or to call squeeze in the test.
> I chose to call squeeze in the test, since we may not always want to
> squeeze.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/arokem/python-matlab-bridge/pull/103/files#r23197883>.
>